### PR TITLE
feat: add openapi spec

### PR DESCRIPTION
**Pull request by kewellcjj**
Added `doc/openapi.yaml`

### CODE REVIEW

**Comment by kewellcjj** [comment]
Screen Shot 2022-05-29 at 2 55 21 PM
Screen Shot 2022-05-29 at 2 58 34 PM

**Comment by awpala** [comment]
> Swagger UI can be accessed at localhost:8082.

We should add this information to the README so it doesn't get lost in the mix...

Also, could you please uncomment port forwarding for `8080` in `/.devcontainer/devcontainer.json` (line 28) accordingly?

Other than those two small things, works well on my end--nice work!

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
 	"forwardPorts": [
 		1927,
 		5432
+		// 8080
```
Comment (awpala) [inline .devcontainer/devcontainer.json]: Woops, just realized this was done deliberately here as part of this PR in the first place lol!

I think we should just clarify in the README specifically how to access the Swagger UI (both locally & via container) so it's clear for other devs who aren't familiar...

Comment (kewellcjj) [reply .devcontainer/devcontainer.json]: Just mapped 8080 to 8080 and add a section in the readme to avoid confusion.

**Review by awpala — DISMISSED** [review]

**Comment by kewellcjj** [comment]
I dropped swagger UI, the openapi.yaml can be viewd in https://redocly.github.io/redoc/, also can be imported to Postman. Also mentioned in README.

**Review by awpala — APPROVED** [review]

**Pull request closed without merging**